Update testing form actions docs with new form-actions key

Mrkbingham opened the pull request:
## Description

Updates docs based on the changed made in https://github.com/filamentphp/filament/pull/17875

## Visual changes

n/a

## Functional changes

- [ x ] Code style has been fixed by running the `composer cs` command.
- [ x ] Changes have been tested to not break existing functionality.
- [ x ] Documentation is up-to-date.

Mrkbingham committed:
updates docs

Mrkbingham committed:
Update 05-testing-actions.md

danharrin reviewed (changes requested):
Can you move this to the "testing resources" section please, in the context of testing custom create/edit page form actions.

Mrkbingham committed:
Document testing create and edit form actions

Added sections for testing create and edit form actions using TestAction in the documentation.

Mrkbingham committed:
Modifies action section to essentially a footnote

Removed example code for testing form actions and updated references to documentation for testing create and edit form actions.

danharrin committed:
Merge branch '4.x' into pr/17887

danharrin committed:
clean up and reduce duplication
